### DISCUSSION TITLE
Lesson 5: How copy problem is solved for data owners?

### DISCUSSION THREAD

**Discussion by IRailean** [OpenMined/courses#222] [Course: Our Privacy Opportunity] [answered]
In Lesson 5: Machine Learning Startups Need Structured Transparency  16:35 Andrew tells that "nobody would have a financial incentive to share a copy of the data anymore". 
As a customer I'm buying this  encrypted data to get some insights, conclusions, etc. I do not need to look at the original data. 
What prevents me to share this encrypted data with other people, who also don't need to look at the original data? 
Encrypted data will work fine for them too. The data owners will lose their money. How data owner can prevent this?

**Comment by ygambhir** [comment, marked as answer]
I think what's important here is the use of Federated Learning / remote computation. It would make sense for the data owner to always own the copy of their data and then give remote access to potential data scientists who want to train models or extract insights. That's at least how I understand it!

**Reply by IRailean** [reply]
Yes, this idea is true. But, as a customer, I receive some information, not the original data, but some information, which I may use as input for my algorithm. As long as it is used as input to my own algorithm, I may also store this information. If I have this information stored (not the original data, but the data that the data owner gave me, encrypted with a bit of randomness, etc.), what prevents me to share it with other customers or even sell it?

**Reply by Nilanshrajput** [reply]
@IRailean it is true you can share the insights you get from your computation over encrypted data, and you can compare this situation with following: a model you trained on Encrypted data, now you can share this model but this doesn't necessarily mean the data have lost its value as someone else may want to train difference model for a different purpose for that he will need to use that encrypted data but yes if a lot of such models are trained the data starts to lose its value as those models could be shared to get an insight of data, now this also brings in the discussion of Privacy Budget as in the course it is discussed Global privacy budget and individual privacy Budget both are present, this privacy budget ensures that not too much information about the data is leaked and data owner can monetize the data accordingly.

**Reply by hershd23** [reply]
@IRailean what you are actually talking about here relates to the privacy budget and @Nilanshrajput correctly points out the sort of balance required to monetize the data correctly. If too much data is leaked the worth of the dataset with the data owner is extinguished. 

There has been academic work in this direction. You can read this [paper](https://arxiv.org/pdf/2003.00973.pdf) if you are interested. This paper models privacy to monetary budget

**Reply by leriomaggio** [reply]
@IRailean I feel you already received quite an extensive list of reply and if you're happy with those, pls feel free to mark this as "Answered" :)

For the sole sake of discussion, I just wanted to share another possible point of view in this data **and** model sharing scenario, as food for thoughts. 

Let's say you're the founder of an innovative startup, who had a brilliant idea of a revolutionary new AI model to solve a very important problem. Now, here is your dilemma: you know your idea and your `wonderful model`® will bring to your startup the income revenue to grow and sustain the company.
However, you need **data** to train and test your model, and there is also another tiny hiccup to solve: all the existing dataset you might need contain sensitive information, and so they are unlikely to be shared publicly. 

So, there is the _federated learning_ approach which someone already mentioned, which is based on the fact that you give your model to the data owner, the data owner runs your model for you and gives your model back to you trained on their `super precious dataset`™ they had. 

There is a problem though: you shared your very key core of your startup to someone else, hoping that this someone won't disclose nor retain this `wonderful model`® of yours. 
This means that this one-directional sharing strategy cannot be sustained, so different approaches are needed: 
`data owners want their data to not be disclosed, whereas startup owners don't want their model to be shared`

(Technical) solutions to this "dilemma" will be presented in the next PrivateAI course 🙌  😉 

I hope this will give you the right motivations to keep up with the course, and to continue your learning journey!

**Reply by IRailean** [reply]
Thank you @Nilanshrajput , @hershd23 , @leriomaggio for your replies.
@leriomaggio correctly stated that the model must be trained on the data owner's side. In other cases as far as I will be training my model on my side, no matter how secure the data was encrypted, if I can use it for my model, I can share it with other model creators, so they can train their models. **!I do not need to restore the original data! **. 
From the other side, I don't want to give my model to data owners. I'm wondering how this dilemma may be solved, and as @leriomaggio pointed out, it will be discussed in the next course.

**Comment by manik-500** [comment]
@IRailean  I think Andrew told this statement in the context of federated learning . Where you will run your algorithm or statistics on the data owner machine .

It’s true you will get some information (insight about the data) but how much .Will it be enough to reproduce the data ?
In federated learning for addressing output privacy when you request for some data(it can be output or intermediate variables ) to see, first it will count how sensitive the data is regarding the input . If it’s too sensitive then it will add more noise to the data to ensure that no extra information is leaked beyond the privacy budget Epsilon .
So if you try to get more sensitive data so that you can reproduce the input data ,then it will add too much noise that you can’t reproduce the input data , you will only get the insight that is intended for your use cases based on the privacy budget Epsilon .
So you can’t reproduce the input data from the output .

Let's talk about your second question .
What's about the information or insight you are getting from the output or from your model.
Can you use it as input for another algorithm or sell it ?
For answering this question let suppose you are running ml algorithm or statistics on grocery data, from that grocery data you want to get the average income of people within a certain region using ml models or statistics . But you can’t infer the health condition of the buyer from the output of your ml models or statistics. On the other hand from the grocery data you can infer the health condition of patients .

Behind the scene differential privacy is actually solving the bundling problem of data . So when you run ml models or statistics on grocery data to get average income , you only get insight about income which is unbundled from health insight . So you can’t reuse the output for another purpose on the other hand original data can .

Please let us know if you have further questions .

**Reply by IRailean** [reply]
Hi @manik-500, thank you for your reply. Maybe I wasn't precise enough in my question. I do not want to restore the original data. I was wondering that if I'm training a model on my side, how data copying (also encrypted data copying) may be prevented. @leriomaggio mentioned that I must run my algorithm on the data owner's machine and that's the only way to solve this problem.

**Reply by leriomaggio** [reply]
Yes @IRailean that's one part of the story - the next step will be to run the model there by using mechanism to prevent either your model to "see" the data (as in "working on raw/plain/un-encrypted data"), as well as the data owner to access *your* model!

In the next course, we will find out how this is possible, and what technology will sustain this new computational paradigm :) 🙌 it will be lots of fun ;)

**Reply by manik-500** [reply]
Hi @IRailean  as Andrew told this statement in the context of federated learning , Your ml algorithm will run on a data owners machine . But the data owners still can copy your ml algorithm. So the copy problem still exists . 
For removing copy problems you can encrypt your ml algorithm with homomorphic encryption so that data owners can’t copy your ml algorithm and you can do your desired computation . Thus the copy problem can be removed .
